In [1]:
import pandas as pd
import numpy as np
import string
import pickle
from sklearn.metrics import classification_report 
from nltk.tokenize import RegexpTokenizer
from nltk import pos_tag
from nltk.corpus import stopwords
from nltk.stem import WordNetLemmatizer
from sklearn.preprocessing import LabelEncoder
from collections import defaultdict
from nltk.corpus import wordnet as wn
from sklearn.feature_extraction.text import TfidfVectorizer
from nltk.stem.snowball import SnowballStemmer
from nltk.stem.porter import PorterStemmer
from sklearn.feature_extraction.text import CountVectorizer
from sklearn import model_selection, naive_bayes, svm
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score
from gensim.models.doc2vec import Doc2Vec, TaggedDocument
from gensim import corpora, models
import gensim
import nltk
from scipy import spatial

nltk.download('punkt')
nltk.download('wordnet')
nltk.download('averaged_perceptron_tagger')
nltk.download('stopwords')


[nltk_data] Downloading package punkt to /home/azmi/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package wordnet to /home/azmi/nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package averaged_perceptron_tagger to
[nltk_data]     /home/azmi/nltk_data...
[nltk_data]   Package averaged_perceptron_tagger is already up-to-
[nltk_data]       date!
[nltk_data] Downloading package stopwords to /home/azmi/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!


True

In [2]:
bugs=pd.read_csv("eclipse_jdt.csv")

In [3]:
bugs.head()

Issue_id Priority Component  Duplicated_issue  \
0      1518       P1     Debug               NaN   
1      1519       P3     Debug               NaN   
2      1520       P3     Debug               NaN   
3      1521       P3     Debug               NaN   
4      1522       P2     Debug            1663.0   

                                               Title  \
0                 Icons needed for actions (1GI5UXW)   
1              README: Hit count not reset (1GET20Y)   
2               Use styled text in console (1G9S1YF)   
3              StringBuffer representation (1GE3BFA)   
4  Feature: use #toString to display variable val...   

                                         Description    Status  Resolution  \
0  JGS (8/8/01 5:20:19 PM); \tWe need enabled; di...  VERIFIED       FIXED   
1  JGS (6/4/01 11:43:47 AM); \tSet a breakpoint i...  RESOLVED  WORKSFORME   
2  DW (2/26/01 5:56:36 PM); \tCould allow users t...    CLOSED     WONTFIX   
3  JGS (5/21/01 2:09:10 PM); \tTo be consistent; ...  VERIFIED       FIXED   
4  DW (9/27/00 11:12:31 AM); \tCurrently; the typ...  RESOLVED   DUPLICATE   

  Version               Created_time              Resolved_time  
0     2.0  2001-10-10 22:14:00 -0400  2001-10-18 11:51:14 -0400  
1     2.0  2001-10-10 22:14:00 -0400  2001-11-28 13:42:46 -0500  
2     2.0  2001-10-10 22:14:00 -0400  2002-06-26 11:32:05 -0400  
3     2.0  2001-10-10 22:14:00 -0400  2002-02-05 10:37:09 -0500  
4     2.0  2001-10-10 22:14:00 -0400  2001-10-17 10:17:47 -0400

In [4]:
bugs['Title_Desc_Final'] = bugs['Title'].astype(str) + " " +bugs['Description'].astype(str)
type(bugs['Title_Desc_Final'][4])

str

In [9]:
def remove_punctuation(text):
    no_punct="".join([c for c in text if c not in string.punctuation])
    return no_punct

tokenizer = RegexpTokenizer(r'\w+')

def remove_stopwords(text):
    words=[w for w in text if w not in stopwords.words('english')]
    return words
lemmatizer = WordNetLemmatizer()
def word_lemmatizer(text):
    lem_text = [lemmatizer.lemmatize(i) for i in text]
    return lem_text
stemmer = PorterStemmer()

def remove_numeric(text):
    words=[w for w in text if not w.isnumeric()]
    return words

def word_stemmer(text):
    stem_text = [stemmer.stem(i) for i in text]
    return stem_text

In [10]:
bugs_test = bugs[bugs['Resolution']=='DUPLICATE']

In [12]:
bugs['Title_Desc_Final']= bugs['Title_Desc_Final'].apply(lambda x: remove_punctuation(x))
bugs['Title_Desc_Final']= bugs['Title_Desc_Final'].apply(lambda x: tokenizer.tokenize(x.lower()))
bugs['Title_Desc_Final']= bugs['Title_Desc_Final'].apply(lambda x: remove_stopwords(x))
bugs['Title_Desc_Final']= bugs['Title_Desc_Final'].apply(lambda x: remove_numeric(x))
bugs['Title_Desc_Final']= bugs['Title_Desc_Final'].apply(lambda x: word_lemmatizer(x))
#bugs['Title_Final']= bugs['Title_Final'].apply(lambda x: word_stemmer(x))

In [13]:
bugs['Title_Desc_Final'].head(20)

0     [icon, needed, action, 1gi5uxw, jgs, pm, need,...
1     [readme, hit, count, reset, 1get20y, jgs, set,...
2     [use, styled, text, console, 1g9s1yf, dw, pm, ...
3     [stringbuffer, representation, 1ge3bfa, jgs, p...
4     [feature, use, tostring, display, variable, va...
5     [breakpoint, invalid, location, 1g4f8p8, d, pm...
6     [feature, displaying, instruction, pointer, 1g...
7     [feature, lock, monitor, 1g3a7zh, dw, support,...
8     [snippet, evaluation, support, import, 1g47213...
9     [scrapbook, snippet, editor, color, snippet, o...
10    [snippet, start, swt, program, twice, 1g47cme,...
11    [exit, allow, user, continue, execution, targe...
12    [ui, inconsistent, gui, behavior, breakpoints,...
13    [feature, navigate, console, output, resource,...
14    [ui, right, click, run, button, 1g5t50t, dw, p...
15    [variable, snapshot, compare, jgs, pm, involve...
16    [dcr, support, debug, demand, 1g71c81, eg, pm,...
17    [performance, optimization, handle, really

In [14]:
bugs

Issue_id Priority Component  Duplicated_issue  \
0          1518       P1     Debug               NaN   
1          1519       P3     Debug               NaN   
2          1520       P3     Debug               NaN   
3          1521       P3     Debug               NaN   
4          1522       P2     Debug            1663.0   
...         ...      ...       ...               ...   
45291    424739       P3      Core               NaN   
45292    424742       P3      Core               NaN   
45293    424772       P3      Text               NaN   
45294    424784       P3        UI               NaN   
45295    424785       P3        UI               NaN   

                                                   Title  \
0                     Icons needed for actions (1GI5UXW)   
1                  README: Hit count not reset (1GET20Y)   
2                   Use styled text in console (1G9S1YF)   
3                  StringBuffer representation (1GE3BFA)   
4      Feature: use #toString to display variable val...   
...                                                  ...   
45291  During my development i am facing lots of prob...   
45292     [1.8] NPE in LambdaExpression.isCompatibleWith   
45293  [typing] Correct Indentation for { is wrong wh...   
45294                                                yes   
45295                eclipse cant show fully code editor   

                                             Description    Status  \
0      JGS (8/8/01 5:20:19 PM); \tWe need enabled; di...  VERIFIED   
1      JGS (6/4/01 11:43:47 AM); \tSet a breakpoint i...  RESOLVED   
2      DW (2/26/01 5:56:36 PM); \tCould allow users t...    CLOSED   
3      JGS (5/21/01 2:09:10 PM); \tTo be consistent; ...  VERIFIED   
4      DW (9/27/00 11:12:31 AM); \tCurrently; the typ...  RESOLVED   
...                                                  ...       ...   
45291  -- Configuration Details --; Product: Eclipse ...  VERIFIED   
45292  package two.test;; ; class TestInlineLambdaArr...  RESOLVED   
45293  Created attachment 238617; Zipped demo project...  RESOLVED   
45294                                                NaN    CLOSED   
45295  Created attachment 238626; Capture screen; ; E...  RESOLVED   

       Resolution Version               Created_time  \
0           FIXED     2.0  2001-10-10 22:14:00 -0400   
1      WORKSFORME     2.0  2001-10-10 22:14:00 -0400   
2         WONTFIX     2.0  2001-10-10 22:14:00 -0400   
3           FIXED     2.0  2001-10-10 22:14:00 -0400   
4       DUPLICATE     2.0  2001-10-10 22:14:00 -0400   
...           ...     ...                        ...   
45291  WORKSFORME     4.4  2013-12-30 02:05:00 -0500   
45292       FIXED     4.4  2013-12-30 04:35:00 -0500   
45293       FIXED     4.4  2013-12-31 03:33:00 -0500   
45294     INVALID   4.3.1  2013-12-31 21:18:00 -0500   
45295     INVALID     4.4  2013-12-31 22:40:00 -0500   

                   Resolved_time  \
0      2001-10-18 11:51:14 -0400   
1      2001-11-28 13:42:46 -0500   
2      2002-06-26 11:32:05 -0400   
3      2002-02-05 10:37:09 -0500   
4      2001-10-17 10:17:47 -0400   
...                          ...   
45291  2014-01-21 04:02:11 -0500   
45292  2014-01-02 08:03:23 -0500   
45293  2014-01-14 10:20:33 -0500   
45294  2014-01-01 01:18:07 -0500   
45295  2014-01-03 04:24:52 -0500   

                                        Title_Desc_Final  
0      [icon, needed, action, 1gi5uxw, jgs, pm, need,...  
1      [readme, hit, count, reset, 1get20y, jgs, set,...  
2      [use, styled, text, console, 1g9s1yf, dw, pm, ...  
3      [stringbuffer, representation, 1ge3bfa, jgs, p...  
4      [feature, use, tostring, display, variable, va...  
...                                                  ...  
45291  [development, facing, lot, problem, using, sho...  
45292  [npe, lambdaexpressioniscompatiblewith, packag...  
45293  [typing, correct, indentation, wrong, declarat...  
45294                                         [yes, nan]  
45295  [eclipse, 

In [15]:
# Convert tokenized document into gensim formated tagged data
tagged_data = [TaggedDocument(d, [i]) for i, d in enumerate(bugs['Title_Desc_Final'])]
tagged_data

[TaggedDocument(words=['icon', 'needed', 'action', '1gi5uxw', 'jgs', 'pm', 'need', 'enabled', 'disabled', 'hover', 'icon', 'following', 'action', 'copytoclipboardactiondelegate', 'relaunchactiondelegate', 'terminateandremoveactiondelegate', 'dw', 'pm', 'use', 'standard', 'copy', 'icon', 'copy', 'clipboard', 'desktop', 'likely', 'expose', 'dw', 'pm', 'made', 'request', 'relaunch', 'terminate', 'terminate', 'remove'], tags=[0]),
 TaggedDocument(words=['readme', 'hit', 'count', 'reset', '1get20y', 'jgs', 'set', 'breakpoint', 'infiniteloop', 'hit', 'count', 'suspended', 'right', 'click', 'breakpoint', 'breakpoints', 'viewer', 'enable', 'hit', 'count', 'stay', 'previous', 'value', 'reset', 'zero', 'dw', 'reenable', 'breakpoint', 'previous', 'hitcount', 'd', 'change', 'hit', 'count', 'breakpoint', 'normal', 'breakpoint', 'set', 'hit', 'count', 'zero', 'likely', 'readme', 'item', 'dw', 'pm', 'make', 'readme', 'dw', 'added', 'readme'], tags=[1]),
 TaggedDocument(words=['use', 'styled', 'text',

In [21]:
model = Doc2Vec(tagged_data, vector_size=300, window=2, min_count=1, workers=4, epochs = 20)
model.save("test_doc2vec.model")
model= Doc2Vec.load("test_doc2vec.model")
model.wv.vocab

{'icon': <gensim.models.keyedvectors.Vocab at 0x7f2417e132d0>,
 'needed': <gensim.models.keyedvectors.Vocab at 0x7f2417e13450>,
 'action': <gensim.models.keyedvectors.Vocab at 0x7f2417e13b90>,
 '1gi5uxw': <gensim.models.keyedvectors.Vocab at 0x7f2417e13c90>,
 'jgs': <gensim.models.keyedvectors.Vocab at 0x7f2417e13d90>,
 'pm': <gensim.models.keyedvectors.Vocab at 0x7f2417e13e90>,
 'need': <gensim.models.keyedvectors.Vocab at 0x7f2417e13f90>,
 'enabled': <gensim.models.keyedvectors.Vocab at 0x7f2417e13190>,
 'disabled': <gensim.models.keyedvectors.Vocab at 0x7f2417e13410>,
 'hover': <gensim.models.keyedvectors.Vocab at 0x7f2417e135d0>,
 'following': <gensim.models.keyedvectors.Vocab at 0x7f2417e13710>,
 'copytoclipboardactiondelegate': <gensim.models.keyedvectors.Vocab at 0x7f2417e13750>,
 'relaunchactiondelegate': <gensim.models.keyedvectors.Vocab at 0x7f2417e13850>,
 'terminateandremoveactiondelegate': <gensim.models.keyedvectors.Vocab at 0x7f2417e13910>,
 'dw': <gensim.models.keyedvec

In [17]:
def get_duplicate_issue_id(dataset, issue_id):
    if str(dataset[dataset["Issue_id"]==issue_id]['Duplicated_issue'].values[0])=='nan':
        return None
    return dataset[dataset["Issue_id"]==issue_id]['Duplicated_issue'].values[0].astype(int)


def is_issue_exist_in_dataset(dataset,issue_id):
    return len(dataset[dataset["Issue_id"]==issue_id]) != 0

def get_issue_id_with_row( row_number):
    return bugs['Issue_id'][row_number]

def get_issue_title_with_id(dataset, issue_id):
    return dataset[dataset["Issue_id"]==issue_id]['Description'].values[0]

def get_issue_title_final_with_id(dataset, issue_id):
    return dataset[dataset["Issue_id"]==issue_id]['Title_Desc_Final'].values[0]

def get_n_most_similar_docs(dataset, issue_id, n):
    test_doc = get_issue_title_final_with_id(dataset, issue_id)
    similars = model.docvecs.most_similar(positive=[model.infer_vector(test_doc)],topn=len(dataset)-1)
 
    issue_list=[]
    for i in range(0,len(similars)-1):
        if is_component_same(dataset, issue_id, get_issue_id_with_row(similars[i][0])) and issue_id != similars[i][0]:
            issue_list.append(get_issue_id_with_row(similars[i][0]))

        if len(issue_list)==n:
            return issue_list
                
    return issue_list

def get_component_with_id(dataset, issue_id):
    return dataset[dataset["Issue_id"]==issue_id]['Component'].values[0]

def is_component_same(dataset, issue1, issue2):
    return get_component_with_id(dataset, issue1) == get_component_with_id(dataset, issue2)

def duplicate_issue_id_exists_in_similarity_list(duplicate_id, similarity_list):
    return duplicate_id in similarity_list

In [ ]:
count=0
count1=0
for item in bugs_test["Issue_id"]:
    if is_issue_exist_in_dataset(bugs, get_duplicate_issue_id(bugs, item)):
        count1=count1+1
        print("Issue id: " + str(item) + " Duplicate issue id: "+ str(get_duplicate_issue_id(bugs, item)))
        print(get_component_with_id(bugs,get_duplicate_issue_id(bugs, item)))
        if is_component_same(bugs, item, get_duplicate_issue_id(bugs, item)):
            count=count+1

In [ ]:
(100*count)/count1

In [ ]:
count

In [ ]:
count1

In [ ]:
(100*count1)/len(bugs_test)

In [22]:
tp=0
fn=0
for issue in bugs_test['Issue_id']:
    print("Issue: " + str(issue)+" Duplicate Issue: " +str(get_duplicate_issue_id(bugs_test, issue)))
    similar_list=get_n_most_similar_docs(bugs, issue, 20)
    if duplicate_issue_id_exists_in_similarity_list(get_duplicate_issue_id(bugs_test,issue),similar_list):
        tp=tp+1
    else:
        fn=fn+1
    

Issue: 1522 Duplicate Issue: 1663
Issue: 1523 Duplicate Issue: 1555
Issue: 1526 Duplicate Issue: 1633
Issue: 1536 Duplicate Issue: 16036
Issue: 1538 Duplicate Issue: 1605
Issue: 1544 Duplicate Issue: 1630
Issue: 1545 Duplicate Issue: 44986
Issue: 1553 Duplicate Issue: 5346
Issue: 1554 Duplicate Issue: 1552
Issue: 1555 Duplicate Issue: 1568
Issue: 1557 Duplicate Issue: 1630
Issue: 1560 Duplicate Issue: 12018
Issue: 1566 Duplicate Issue: 1532
Issue: 1568 Duplicate Issue: 1570
Issue: 1570 Duplicate Issue: 1619
Issue: 1575 Duplicate Issue: 1524
Issue: 1586 Duplicate Issue: 1552
Issue: 1594 Duplicate Issue: 7552
Issue: 1598 Duplicate Issue: 1585
Issue: 1604 Duplicate Issue: 4383
Issue: 1606 Duplicate Issue: 5346
Issue: 1608 Duplicate Issue: 5163
Issue: 1612 Duplicate Issue: 6185
Issue: 1616 Duplicate Issue: 1546
Issue: 1618 Duplicate Issue: 6509
Issue: 1619 Duplicate Issue: 1667
Issue: 1620 Duplicate Issue: 1546
Issue: 1622 Duplicate Issue: 1524
Issue: 1634 Duplicate Issue: 1585
Issue: 1639

Issue: 6045 Duplicate Issue: 3420
Issue: 6048 Duplicate Issue: 5761
Issue: 6053 Duplicate Issue: 6058
Issue: 6056 Duplicate Issue: 6057
Issue: 6070 Duplicate Issue: 5852
Issue: 6084 Duplicate Issue: 1720
Issue: 6085 Duplicate Issue: 9151
Issue: 6094 Duplicate Issue: 3320
Issue: 6120 Duplicate Issue: 3625
Issue: 6128 Duplicate Issue: 5789
Issue: 6129 Duplicate Issue: 5761
Issue: 6136 Duplicate Issue: 6325
Issue: 6145 Duplicate Issue: 9151
Issue: 6150 Duplicate Issue: 5875
Issue: 6160 Duplicate Issue: 6151
Issue: 6161 Duplicate Issue: 1753
Issue: 6165 Duplicate Issue: 13905
Issue: 6167 Duplicate Issue: 9185
Issue: 6176 Duplicate Issue: 6151
Issue: 6256 Duplicate Issue: 6241
Issue: 6280 Duplicate Issue: 4134
Issue: 6291 Duplicate Issue: 1635
Issue: 6294 Duplicate Issue: 6160
Issue: 6301 Duplicate Issue: 3331
Issue: 6338 Duplicate Issue: 3847
Issue: 6339 Duplicate Issue: 6302
Issue: 6398 Duplicate Issue: 5835
Issue: 6417 Duplicate Issue: 11403
Issue: 6455 Duplicate Issue: 15262
Issue: 6468

Issue: 9726 Duplicate Issue: 9476
Issue: 9766 Duplicate Issue: 9732
Issue: 9785 Duplicate Issue: 7485
Issue: 9843 Duplicate Issue: 13113
Issue: 9853 Duplicate Issue: 7638
Issue: 9880 Duplicate Issue: 1546
Issue: 9888 Duplicate Issue: 9479
Issue: 9890 Duplicate Issue: 13034
Issue: 9897 Duplicate Issue: 18994
Issue: 9925 Duplicate Issue: 22199
Issue: 9939 Duplicate Issue: 9914
Issue: 9949 Duplicate Issue: 5761
Issue: 9951 Duplicate Issue: 12740
Issue: 10029 Duplicate Issue: 6454
Issue: 10041 Duplicate Issue: 6949
Issue: 10045 Duplicate Issue: 11024
Issue: 10048 Duplicate Issue: 9909
Issue: 10051 Duplicate Issue: 9798
Issue: 10097 Duplicate Issue: 5646
Issue: 10129 Duplicate Issue: 7605
Issue: 10137 Duplicate Issue: 9673
Issue: 10151 Duplicate Issue: 10096
Issue: 10166 Duplicate Issue: 9985
Issue: 10172 Duplicate Issue: 9017
Issue: 10202 Duplicate Issue: 10198
Issue: 10219 Duplicate Issue: 8772
Issue: 10228 Duplicate Issue: 5761
Issue: 10237 Duplicate Issue: 8919
Issue: 10238 Duplicate Is

Issue: 13421 Duplicate Issue: 12823
Issue: 13437 Duplicate Issue: 13323
Issue: 13462 Duplicate Issue: 13684
Issue: 13468 Duplicate Issue: 13393
Issue: 13487 Duplicate Issue: 13355
Issue: 13497 Duplicate Issue: 89854
Issue: 13508 Duplicate Issue: 13336
Issue: 13522 Duplicate Issue: 13355
Issue: 13533 Duplicate Issue: 13512
Issue: 13538 Duplicate Issue: 14597
Issue: 13604 Duplicate Issue: 13865
Issue: 13612 Duplicate Issue: 12970
Issue: 13619 Duplicate Issue: 12970
Issue: 13627 Duplicate Issue: 7004
Issue: 13713 Duplicate Issue: 14045
Issue: 13742 Duplicate Issue: 3733
Issue: 13765 Duplicate Issue: 36718
Issue: 13827 Duplicate Issue: 10169
Issue: 13841 Duplicate Issue: 13834
Issue: 13845 Duplicate Issue: 7245
Issue: 13852 Duplicate Issue: 11410
Issue: 13855 Duplicate Issue: 9355
Issue: 13863 Duplicate Issue: 13865
Issue: 13872 Duplicate Issue: 12561
Issue: 13881 Duplicate Issue: 10169
Issue: 13889 Duplicate Issue: 13581
Issue: 13911 Duplicate Issue: 13909
Issue: 13958 Duplicate Issue: 35

Issue: 17723 Duplicate Issue: 19721
Issue: 17726 Duplicate Issue: 17791
Issue: 17727 Duplicate Issue: 17491
Issue: 17732 Duplicate Issue: 19721
Issue: 17750 Duplicate Issue: 12729
Issue: 17797 Duplicate Issue: 16761
Issue: 17798 Duplicate Issue: 16170
Issue: 17804 Duplicate Issue: 17453
Issue: 17824 Duplicate Issue: 19349
Issue: 17827 Duplicate Issue: 16761
Issue: 17856 Duplicate Issue: 14061
Issue: 17866 Duplicate Issue: 15212
Issue: 17888 Duplicate Issue: 16534
Issue: 17949 Duplicate Issue: 17512
Issue: 17950 Duplicate Issue: 16638
Issue: 17953 Duplicate Issue: 16761
Issue: 18011 Duplicate Issue: 15170
Issue: 18012 Duplicate Issue: 18011
Issue: 18013 Duplicate Issue: 17095
Issue: 18014 Duplicate Issue: 18079
Issue: 18022 Duplicate Issue: 1609
Issue: 18070 Duplicate Issue: 16761
Issue: 18084 Duplicate Issue: 11077
Issue: 18098 Duplicate Issue: 19581
Issue: 18107 Duplicate Issue: 17868
Issue: 18113 Duplicate Issue: 15212
Issue: 18171 Duplicate Issue: 14555
Issue: 18203 Duplicate Issue:

Issue: 21359 Duplicate Issue: 16388
Issue: 21381 Duplicate Issue: 19738
Issue: 21388 Duplicate Issue: 21111
Issue: 21393 Duplicate Issue: 34203
Issue: 21394 Duplicate Issue: 21392
Issue: 21396 Duplicate Issue: 18164
Issue: 21408 Duplicate Issue: 8441
Issue: 21441 Duplicate Issue: 21443
Issue: 21442 Duplicate Issue: 21443
Issue: 21459 Duplicate Issue: 15247
Issue: 21474 Duplicate Issue: 41421
Issue: 21482 Duplicate Issue: 21447
Issue: 21496 Duplicate Issue: 1631
Issue: 21515 Duplicate Issue: 21516
Issue: 21537 Duplicate Issue: 23733
Issue: 21540 Duplicate Issue: 21311
Issue: 21558 Duplicate Issue: 32680
Issue: 21576 Duplicate Issue: 21516
Issue: 21588 Duplicate Issue: 21514
Issue: 21629 Duplicate Issue: 18977
Issue: 21680 Duplicate Issue: 21170
Issue: 21689 Duplicate Issue: 20874
Issue: 21705 Duplicate Issue: 7154
Issue: 21709 Duplicate Issue: 19784
Issue: 21725 Duplicate Issue: 21170
Issue: 21735 Duplicate Issue: 21514
Issue: 21759 Duplicate Issue: 19056
Issue: 21765 Duplicate Issue: 2

Issue: 25555 Duplicate Issue: 8518
Issue: 25568 Duplicate Issue: 26870
Issue: 25573 Duplicate Issue: 25549
Issue: 25584 Duplicate Issue: 3979
Issue: 25589 Duplicate Issue: 25585
Issue: 25602 Duplicate Issue: 25044
Issue: 25666 Duplicate Issue: 25597
Issue: 25672 Duplicate Issue: 24957
Issue: 25674 Duplicate Issue: 21402
Issue: 25697 Duplicate Issue: 25689
Issue: 25716 Duplicate Issue: 26027
Issue: 25743 Duplicate Issue: 12144
Issue: 25744 Duplicate Issue: 25743
Issue: 25760 Duplicate Issue: 25230
Issue: 25769 Duplicate Issue: 25242
Issue: 25771 Duplicate Issue: 25832
Issue: 25773 Duplicate Issue: 25113
Issue: 25786 Duplicate Issue: 28812
Issue: 25809 Duplicate Issue: 45179
Issue: 25812 Duplicate Issue: 25900
Issue: 25822 Duplicate Issue: 25875
Issue: 25836 Duplicate Issue: 25701
Issue: 25838 Duplicate Issue: 21140
Issue: 25868 Duplicate Issue: 22969
Issue: 25871 Duplicate Issue: 27231
Issue: 25875 Duplicate Issue: 7224
Issue: 25883 Duplicate Issue: 19738
Issue: 25891 Duplicate Issue: 2

Issue: 29518 Duplicate Issue: 17549
Issue: 29540 Duplicate Issue: 29524
Issue: 29547 Duplicate Issue: 29151
Issue: 29549 Duplicate Issue: 29151
Issue: 29556 Duplicate Issue: 3331
Issue: 29588 Duplicate Issue: 28890
Issue: 29589 Duplicate Issue: 10053
Issue: 29594 Duplicate Issue: 28054
Issue: 29600 Duplicate Issue: 47509
Issue: 29629 Duplicate Issue: 34065
Issue: 29630 Duplicate Issue: 26825
Issue: 29688 Duplicate Issue: 29028
Issue: 29690 Duplicate Issue: 29585
Issue: 29716 Duplicate Issue: 23134
Issue: 29727 Duplicate Issue: 24966
Issue: 29735 Duplicate Issue: 29549
Issue: 29749 Duplicate Issue: 6224
Issue: 29803 Duplicate Issue: 29761
Issue: 29845 Duplicate Issue: 108071
Issue: 29868 Duplicate Issue: 28796
Issue: 29882 Duplicate Issue: 28796
Issue: 29903 Duplicate Issue: 75603
Issue: 29957 Duplicate Issue: 22242
Issue: 29971 Duplicate Issue: 10035
Issue: 30008 Duplicate Issue: 24045
Issue: 30023 Duplicate Issue: 29151
Issue: 30036 Duplicate Issue: 21904
Issue: 30052 Duplicate Issue:

Issue: 33104 Duplicate Issue: 32767
Issue: 33105 Duplicate Issue: 41676
Issue: 33146 Duplicate Issue: 32469
Issue: 33204 Duplicate Issue: 32854
Issue: 33232 Duplicate Issue: 33241
Issue: 33242 Duplicate Issue: 32633
Issue: 33255 Duplicate Issue: 33261
Issue: 33256 Duplicate Issue: 23056
Issue: 33267 Duplicate Issue: 33268
Issue: 33268 Duplicate Issue: 32858
Issue: 33285 Duplicate Issue: 32829
Issue: 33291 Duplicate Issue: 32494
Issue: 33295 Duplicate Issue: 32494
Issue: 33307 Duplicate Issue: 30277
Issue: 33382 Duplicate Issue: 37394
Issue: 33383 Duplicate Issue: 33521
Issue: 33384 Duplicate Issue: 32528
Issue: 33408 Duplicate Issue: 29964
Issue: 33446 Duplicate Issue: 32633
Issue: 33457 Duplicate Issue: 33458
Issue: 33461 Duplicate Issue: 33462
Issue: 33466 Duplicate Issue: 33462
Issue: 33505 Duplicate Issue: 32342
Issue: 33507 Duplicate Issue: 32633
Issue: 33526 Duplicate Issue: 32633
Issue: 33541 Duplicate Issue: 33187
Issue: 33559 Duplicate Issue: 32831
Issue: 33563 Duplicate Issue

Issue: 37434 Duplicate Issue: 22976
Issue: 37437 Duplicate Issue: 24378
Issue: 37452 Duplicate Issue: 37496
Issue: 37453 Duplicate Issue: 37496
Issue: 37466 Duplicate Issue: 133277
Issue: 37469 Duplicate Issue: 19036
Issue: 37481 Duplicate Issue: 12673
Issue: 37496 Duplicate Issue: 154109
Issue: 37511 Duplicate Issue: 37509
Issue: 37512 Duplicate Issue: 12330
Issue: 37536 Duplicate Issue: 3462
Issue: 37538 Duplicate Issue: 23064
Issue: 37548 Duplicate Issue: 36307
Issue: 37568 Duplicate Issue: 38382
Issue: 37585 Duplicate Issue: 21782
Issue: 37609 Duplicate Issue: 3181
Issue: 37637 Duplicate Issue: 34857
Issue: 37638 Duplicate Issue: 36538
Issue: 37641 Duplicate Issue: 37640
Issue: 37725 Duplicate Issue: 36938
Issue: 37749 Duplicate Issue: 39736
Issue: 37757 Duplicate Issue: 48242
Issue: 37764 Duplicate Issue: 37763
Issue: 37769 Duplicate Issue: 35746
Issue: 37804 Duplicate Issue: 36854
Issue: 37805 Duplicate Issue: 37804
Issue: 37811 Duplicate Issue: 37218
Issue: 37817 Duplicate Issue

Issue: 41407 Duplicate Issue: 38797
Issue: 41423 Duplicate Issue: 30315
Issue: 41425 Duplicate Issue: 39781
Issue: 41450 Duplicate Issue: 37513
Issue: 41461 Duplicate Issue: 37937
Issue: 41462 Duplicate Issue: 37937
Issue: 41503 Duplicate Issue: 41459
Issue: 41511 Duplicate Issue: 39205
Issue: 41516 Duplicate Issue: 40476
Issue: 41525 Duplicate Issue: 8748
Issue: 41533 Duplicate Issue: 41534
Issue: 41567 Duplicate Issue: 41488
Issue: 41586 Duplicate Issue: 21086
Issue: 41607 Duplicate Issue: 38082
Issue: 41612 Duplicate Issue: 41028
Issue: 41617 Duplicate Issue: 41651
Issue: 41640 Duplicate Issue: 41421
Issue: 41642 Duplicate Issue: 40995
Issue: 41648 Duplicate Issue: 9355
Issue: 41711 Duplicate Issue: 40995
Issue: 41722 Duplicate Issue: 41721
Issue: 41811 Duplicate Issue: 41255
Issue: 41813 Duplicate Issue: 41953
Issue: 41818 Duplicate Issue: 41231
Issue: 41821 Duplicate Issue: 36479
Issue: 41847 Duplicate Issue: 41823
Issue: 41853 Duplicate Issue: 41190
Issue: 41859 Duplicate Issue: 

Issue: 45582 Duplicate Issue: 44721
Issue: 45586 Duplicate Issue: 45352
Issue: 45590 Duplicate Issue: 45589
Issue: 45591 Duplicate Issue: 45589
Issue: 45622 Duplicate Issue: 45446
Issue: 45639 Duplicate Issue: 21892
Issue: 45645 Duplicate Issue: 45452
Issue: 45659 Duplicate Issue: 25835
Issue: 45742 Duplicate Issue: 24319
Issue: 45747 Duplicate Issue: 45777
Issue: 45779 Duplicate Issue: 51075
Issue: 45791 Duplicate Issue: 45785
Issue: 45819 Duplicate Issue: 43986
Issue: 45830 Duplicate Issue: 36032
Issue: 45891 Duplicate Issue: 45887
Issue: 45900 Duplicate Issue: 41867
Issue: 45919 Duplicate Issue: 44690
Issue: 45925 Duplicate Issue: 38110
Issue: 45941 Duplicate Issue: 43483
Issue: 45959 Duplicate Issue: 45785
Issue: 45961 Duplicate Issue: 12746
Issue: 45964 Duplicate Issue: 25963
Issue: 45970 Duplicate Issue: 45194
Issue: 46009 Duplicate Issue: 45785
Issue: 46018 Duplicate Issue: 10712
Issue: 46023 Duplicate Issue: 46033
Issue: 46041 Duplicate Issue: 46037
Issue: 46087 Duplicate Issue

Issue: 50187 Duplicate Issue: 40038
Issue: 50209 Duplicate Issue: 50207
Issue: 50253 Duplicate Issue: 51075
Issue: 50255 Duplicate Issue: 71244
Issue: 50263 Duplicate Issue: 49265
Issue: 50305 Duplicate Issue: 19543
Issue: 50319 Duplicate Issue: 18338
Issue: 50330 Duplicate Issue: 50331
Issue: 50370 Duplicate Issue: 63040
Issue: 50388 Duplicate Issue: 49686
Issue: 50411 Duplicate Issue: 11624
Issue: 50474 Duplicate Issue: 50277
Issue: 50507 Duplicate Issue: 49673
Issue: 50510 Duplicate Issue: 50679
Issue: 50559 Duplicate Issue: 50560
Issue: 50688 Duplicate Issue: 46918
Issue: 50717 Duplicate Issue: 20042
Issue: 50730 Duplicate Issue: 50313
Issue: 50733 Duplicate Issue: 39720
Issue: 50780 Duplicate Issue: 51516
Issue: 50801 Duplicate Issue: 43483
Issue: 50830 Duplicate Issue: 50780
Issue: 50845 Duplicate Issue: 49298
Issue: 50849 Duplicate Issue: 47396
Issue: 50850 Duplicate Issue: 19771
Issue: 50853 Duplicate Issue: 50849
Issue: 50881 Duplicate Issue: 50880
Issue: 50891 Duplicate Issue

Issue: 55375 Duplicate Issue: 49686
Issue: 55396 Duplicate Issue: 54993
Issue: 55427 Duplicate Issue: 56704
Issue: 55480 Duplicate Issue: 47815
Issue: 55535 Duplicate Issue: 5858
Issue: 55560 Duplicate Issue: 55731
Issue: 55601 Duplicate Issue: 10745
Issue: 55610 Duplicate Issue: 58684
Issue: 55623 Duplicate Issue: 55589
Issue: 55638 Duplicate Issue: 51533
Issue: 55643 Duplicate Issue: 51533
Issue: 55644 Duplicate Issue: 49686
Issue: 55697 Duplicate Issue: 55651
Issue: 55703 Duplicate Issue: 54885
Issue: 55711 Duplicate Issue: 55675
Issue: 55736 Duplicate Issue: 55629
Issue: 55753 Duplicate Issue: 53869
Issue: 55769 Duplicate Issue: 54956
Issue: 55776 Duplicate Issue: 55975
Issue: 55786 Duplicate Issue: 55675
Issue: 55809 Duplicate Issue: 52115
Issue: 55842 Duplicate Issue: 27926
Issue: 55868 Duplicate Issue: 55867
Issue: 55871 Duplicate Issue: 55675
Issue: 55874 Duplicate Issue: 50735
Issue: 55880 Duplicate Issue: 55930
Issue: 55893 Duplicate Issue: 55867
Issue: 55907 Duplicate Issue:

Issue: 61094 Duplicate Issue: 49844
Issue: 61100 Duplicate Issue: 75333
Issue: 61134 Duplicate Issue: 61405
Issue: 61135 Duplicate Issue: 10412
Issue: 61142 Duplicate Issue: 59638
Issue: 61205 Duplicate Issue: 59420
Issue: 61229 Duplicate Issue: 61095
Issue: 61258 Duplicate Issue: 52834
Issue: 61311 Duplicate Issue: 61446
Issue: 61333 Duplicate Issue: 46315
Issue: 61354 Duplicate Issue: 58554
Issue: 61361 Duplicate Issue: 54072
Issue: 61362 Duplicate Issue: 53899
Issue: 61364 Duplicate Issue: 61095
Issue: 61380 Duplicate Issue: 61361
Issue: 61405 Duplicate Issue: 57763
Issue: 61481 Duplicate Issue: 57419
Issue: 61487 Duplicate Issue: 61345
Issue: 61499 Duplicate Issue: 61725
Issue: 61510 Duplicate Issue: 20793
Issue: 61511 Duplicate Issue: 198074
Issue: 61559 Duplicate Issue: 61383
Issue: 61599 Duplicate Issue: 44141
Issue: 61695 Duplicate Issue: 24804
Issue: 61722 Duplicate Issue: 66426
Issue: 61746 Duplicate Issue: 61128
Issue: 61750 Duplicate Issue: 40038
Issue: 61752 Duplicate Issu

Issue: 65999 Duplicate Issue: 64136
Issue: 66050 Duplicate Issue: 65985
Issue: 66076 Duplicate Issue: 64672
Issue: 66152 Duplicate Issue: 65870
Issue: 66229 Duplicate Issue: 66218
Issue: 66248 Duplicate Issue: 62757
Issue: 66257 Duplicate Issue: 63573
Issue: 66262 Duplicate Issue: 66218
Issue: 66298 Duplicate Issue: 67500
Issue: 66310 Duplicate Issue: 66309
Issue: 66311 Duplicate Issue: 66309
Issue: 66452 Duplicate Issue: 84770
Issue: 66508 Duplicate Issue: 66353
Issue: 66528 Duplicate Issue: 64646
Issue: 66555 Duplicate Issue: 64672
Issue: 66587 Duplicate Issue: 45095
Issue: 66592 Duplicate Issue: 64672
Issue: 66594 Duplicate Issue: 66551
Issue: 66627 Duplicate Issue: 25158
Issue: 66628 Duplicate Issue: 64672
Issue: 66642 Duplicate Issue: 67183
Issue: 66649 Duplicate Issue: 52921
Issue: 66655 Duplicate Issue: 69536
Issue: 66662 Duplicate Issue: 64134
Issue: 66668 Duplicate Issue: 67183
Issue: 66671 Duplicate Issue: 66632
Issue: 66676 Duplicate Issue: 60385
Issue: 66703 Duplicate Issue

Issue: 71439 Duplicate Issue: 64749
Issue: 71459 Duplicate Issue: 52921
Issue: 71468 Duplicate Issue: 71195
Issue: 71485 Duplicate Issue: 10104
Issue: 71505 Duplicate Issue: 49686
Issue: 71536 Duplicate Issue: 67969
Issue: 71555 Duplicate Issue: 68981
Issue: 71574 Duplicate Issue: 62605
Issue: 71575 Duplicate Issue: 418692
Issue: 71613 Duplicate Issue: 71016
Issue: 71616 Duplicate Issue: 45333
Issue: 71620 Duplicate Issue: 50690
Issue: 71621 Duplicate Issue: 44764
Issue: 71629 Duplicate Issue: 73389
Issue: 71641 Duplicate Issue: 67969
Issue: 71662 Duplicate Issue: 74035
Issue: 71676 Duplicate Issue: 71673
Issue: 71704 Duplicate Issue: 71016
Issue: 71749 Duplicate Issue: 39661
Issue: 71758 Duplicate Issue: 63692
Issue: 71762 Duplicate Issue: 48231
Issue: 71774 Duplicate Issue: 59171
Issue: 71797 Duplicate Issue: 75701
Issue: 71842 Duplicate Issue: 31559
Issue: 71851 Duplicate Issue: 67395
Issue: 71855 Duplicate Issue: 34552
Issue: 71856 Duplicate Issue: 71767
Issue: 71863 Duplicate Issu

Issue: 77078 Duplicate Issue: 73963
Issue: 77103 Duplicate Issue: 77030
Issue: 77113 Duplicate Issue: 182537
Issue: 77115 Duplicate Issue: 76165
Issue: 77126 Duplicate Issue: 77030
Issue: 77139 Duplicate Issue: 6038
Issue: 77142 Duplicate Issue: 76313
Issue: 77211 Duplicate Issue: 78916
Issue: 77224 Duplicate Issue: 28627
Issue: 77237 Duplicate Issue: 74673
Issue: 77248 Duplicate Issue: 63366
Issue: 77324 Duplicate Issue: 76880
Issue: 77334 Duplicate Issue: 77033
Issue: 77360 Duplicate Issue: 78266
Issue: 77374 Duplicate Issue: 173122
Issue: 77432 Duplicate Issue: 74505
Issue: 77445 Duplicate Issue: 10104
Issue: 77452 Duplicate Issue: 46696
Issue: 77473 Duplicate Issue: 117758
Issue: 77475 Duplicate Issue: 41421
Issue: 77525 Duplicate Issue: 42086
Issue: 77580 Duplicate Issue: 76960
Issue: 77590 Duplicate Issue: 77548
Issue: 77594 Duplicate Issue: 77034
Issue: 77616 Duplicate Issue: 54943
Issue: 77619 Duplicate Issue: 77622
Issue: 77655 Duplicate Issue: 77894
Issue: 77669 Duplicate Iss

Issue: 82225 Duplicate Issue: 79849
Issue: 82269 Duplicate Issue: 75600
Issue: 82275 Duplicate Issue: 89739
Issue: 82287 Duplicate Issue: 81824
Issue: 82288 Duplicate Issue: 78591
Issue: 82303 Duplicate Issue: 78358
Issue: 82319 Duplicate Issue: 81945
Issue: 82420 Duplicate Issue: 82337
Issue: 82421 Duplicate Issue: 76890
Issue: 82427 Duplicate Issue: 82394
Issue: 82448 Duplicate Issue: 81825
Issue: 82455 Duplicate Issue: 81724
Issue: 82464 Duplicate Issue: 82830
Issue: 82474 Duplicate Issue: 81825
Issue: 82484 Duplicate Issue: 82485
Issue: 82487 Duplicate Issue: 75490
Issue: 82497 Duplicate Issue: 81719
Issue: 82511 Duplicate Issue: 81618
Issue: 82521 Duplicate Issue: 90082
Issue: 82534 Duplicate Issue: 75074
Issue: 82538 Duplicate Issue: 80221
Issue: 82551 Duplicate Issue: 76326
Issue: 82554 Duplicate Issue: 79112
Issue: 82563 Duplicate Issue: 75570
Issue: 82584 Duplicate Issue: 82005
Issue: 82687 Duplicate Issue: 86971
Issue: 82699 Duplicate Issue: 73969
Issue: 82704 Duplicate Issue

Issue: 88506 Duplicate Issue: 86291
Issue: 88551 Duplicate Issue: 86078
Issue: 88570 Duplicate Issue: 49412
Issue: 88576 Duplicate Issue: 86177
Issue: 88581 Duplicate Issue: 87042
Issue: 88587 Duplicate Issue: 54968
Issue: 88594 Duplicate Issue: 82494
Issue: 88616 Duplicate Issue: 41867
Issue: 88628 Duplicate Issue: 100135
Issue: 88744 Duplicate Issue: 78416
Issue: 88745 Duplicate Issue: 89333
Issue: 88765 Duplicate Issue: 88892
Issue: 88812 Duplicate Issue: 84455
Issue: 88825 Duplicate Issue: 89333
Issue: 88855 Duplicate Issue: 89932
Issue: 88928 Duplicate Issue: 88941
Issue: 88974 Duplicate Issue: 88973
Issue: 88976 Duplicate Issue: 82494
Issue: 88980 Duplicate Issue: 22285
Issue: 89002 Duplicate Issue: 82850
Issue: 89047 Duplicate Issue: 95909
Issue: 89058 Duplicate Issue: 88993
Issue: 89060 Duplicate Issue: 87955
Issue: 89064 Duplicate Issue: 83937
Issue: 89182 Duplicate Issue: 85413
Issue: 89186 Duplicate Issue: 87868
Issue: 89236 Duplicate Issue: 89240
Issue: 89266 Duplicate Issu

Issue: 95604 Duplicate Issue: 35867
Issue: 95616 Duplicate Issue: 45095
Issue: 95626 Duplicate Issue: 84324
Issue: 95649 Duplicate Issue: 149638
Issue: 95668 Duplicate Issue: 91678
Issue: 95686 Duplicate Issue: 95326
Issue: 95699 Duplicate Issue: 38185
Issue: 95713 Duplicate Issue: 95679
Issue: 95723 Duplicate Issue: 93941
Issue: 95725 Duplicate Issue: 95399
Issue: 95728 Duplicate Issue: 82254
Issue: 95837 Duplicate Issue: 60385
Issue: 95898 Duplicate Issue: 89935
Issue: 95904 Duplicate Issue: 95142
Issue: 95917 Duplicate Issue: 92799
Issue: 95925 Duplicate Issue: 91543
Issue: 95936 Duplicate Issue: 83389
Issue: 95951 Duplicate Issue: 96974
Issue: 95966 Duplicate Issue: 95975
Issue: 95975 Duplicate Issue: 95976
Issue: 95976 Duplicate Issue: 95977
Issue: 96007 Duplicate Issue: 82169
Issue: 96038 Duplicate Issue: 64665
Issue: 96049 Duplicate Issue: 96048
Issue: 96072 Duplicate Issue: 82850
Issue: 96077 Duplicate Issue: 144469
Issue: 96093 Duplicate Issue: 96094
Issue: 96094 Duplicate Iss

Issue: 102477 Duplicate Issue: 96234
Issue: 102483 Duplicate Issue: 101863
Issue: 102490 Duplicate Issue: 71987
Issue: 102507 Duplicate Issue: 102506
Issue: 102545 Duplicate Issue: 101456
Issue: 102551 Duplicate Issue: 91327
Issue: 102569 Duplicate Issue: 53727
Issue: 102579 Duplicate Issue: 71987
Issue: 102594 Duplicate Issue: 101456
Issue: 102606 Duplicate Issue: 102608
Issue: 102609 Duplicate Issue: 102615
Issue: 102646 Duplicate Issue: 41392
Issue: 102651 Duplicate Issue: 53451
Issue: 102722 Duplicate Issue: 92571
Issue: 102735 Duplicate Issue: 87868
Issue: 102784 Duplicate Issue: 52115
Issue: 102812 Duplicate Issue: 65841
Issue: 102818 Duplicate Issue: 68144
Issue: 102844 Duplicate Issue: 221767
Issue: 102934 Duplicate Issue: 102102
Issue: 102939 Duplicate Issue: 24804
Issue: 102976 Duplicate Issue: 87522
Issue: 103005 Duplicate Issue: 101247
Issue: 103018 Duplicate Issue: 44894
Issue: 103046 Duplicate Issue: 85881
Issue: 103095 Duplicate Issue: 9503
Issue: 103114 Duplicate Issue:

Issue: 111032 Duplicate Issue: 122620
Issue: 111101 Duplicate Issue: 67705
Issue: 111163 Duplicate Issue: 110943
Issue: 111215 Duplicate Issue: 110336
Issue: 111232 Duplicate Issue: 75519
Issue: 111261 Duplicate Issue: 44627
Issue: 111270 Duplicate Issue: 102780
Issue: 111336 Duplicate Issue: 110206
Issue: 111443 Duplicate Issue: 71987
Issue: 111447 Duplicate Issue: 147516
Issue: 111463 Duplicate Issue: 110643
Issue: 111464 Duplicate Issue: 110650
Issue: 111506 Duplicate Issue: 40282
Issue: 111577 Duplicate Issue: 111558
Issue: 111585 Duplicate Issue: 111586
Issue: 111650 Duplicate Issue: 111573
Issue: 111737 Duplicate Issue: 92893
Issue: 111792 Duplicate Issue: 110346
Issue: 111799 Duplicate Issue: 92264
Issue: 111837 Duplicate Issue: 60385
Issue: 111838 Duplicate Issue: 42593
Issue: 111937 Duplicate Issue: 168520
Issue: 112003 Duplicate Issue: 101080
Issue: 112053 Duplicate Issue: 107831
Issue: 112219 Duplicate Issue: 33936
Issue: 112260 Duplicate Issue: 67562
Issue: 112261 Duplicate

Issue: 122983 Duplicate Issue: 119161
Issue: 122991 Duplicate Issue: 103381
Issue: 123013 Duplicate Issue: 56062
Issue: 123017 Duplicate Issue: 122826
Issue: 123033 Duplicate Issue: 123118
Issue: 123196 Duplicate Issue: 122176
Issue: 123226 Duplicate Issue: 210567
Issue: 123321 Duplicate Issue: 44894
Issue: 123390 Duplicate Issue: 112268
Issue: 123394 Duplicate Issue: 89428
Issue: 123402 Duplicate Issue: 123118
Issue: 123518 Duplicate Issue: 111270
Issue: 123519 Duplicate Issue: 186192
Issue: 123528 Duplicate Issue: 87299
Issue: 123529 Duplicate Issue: 99137
Issue: 123580 Duplicate Issue: 95708
Issue: 123592 Duplicate Issue: 39065
Issue: 123604 Duplicate Issue: 120615
Issue: 123639 Duplicate Issue: 123479
Issue: 123753 Duplicate Issue: 120135
Issue: 123780 Duplicate Issue: 123782
Issue: 123839 Duplicate Issue: 110579
Issue: 123898 Duplicate Issue: 38185
Issue: 123912 Duplicate Issue: 53650
Issue: 123944 Duplicate Issue: 58699
Issue: 124119 Duplicate Issue: 95056
Issue: 124148 Duplicate

Issue: 133803 Duplicate Issue: 120088
Issue: 133804 Duplicate Issue: 133668
Issue: 133817 Duplicate Issue: 133348
Issue: 133892 Duplicate Issue: 134090
Issue: 134035 Duplicate Issue: 134034
Issue: 134061 Duplicate Issue: 133348
Issue: 134070 Duplicate Issue: 119025
Issue: 134226 Duplicate Issue: 40623
Issue: 134293 Duplicate Issue: 134408
Issue: 134338 Duplicate Issue: 134536
Issue: 134439 Duplicate Issue: 95839
Issue: 134476 Duplicate Issue: 134839
Issue: 134514 Duplicate Issue: 81344
Issue: 134625 Duplicate Issue: 122247
Issue: 134684 Duplicate Issue: 171075
Issue: 134761 Duplicate Issue: 110594
Issue: 134802 Duplicate Issue: 134803
Issue: 134840 Duplicate Issue: 140257
Issue: 134842 Duplicate Issue: 134689
Issue: 134844 Duplicate Issue: 134746
Issue: 135081 Duplicate Issue: 133594
Issue: 135115 Duplicate Issue: 44419
Issue: 135166 Duplicate Issue: 112774
Issue: 135225 Duplicate Issue: 133348
Issue: 135263 Duplicate Issue: 132671
Issue: 135264 Duplicate Issue: 134689
Issue: 135277 Du

Issue: 147880 Duplicate Issue: 147875
Issue: 147943 Duplicate Issue: 138877
Issue: 148243 Duplicate Issue: 182179
Issue: 148249 Duplicate Issue: 140561
Issue: 148261 Duplicate Issue: 148015
Issue: 148263 Duplicate Issue: 150121
Issue: 148387 Duplicate Issue: 92264
Issue: 148630 Duplicate Issue: 148631
Issue: 148657 Duplicate Issue: 176210
Issue: 148795 Duplicate Issue: 148247
Issue: 148812 Duplicate Issue: 138877
Issue: 148921 Duplicate Issue: 49619
Issue: 148968 Duplicate Issue: 119857
Issue: 149042 Duplicate Issue: 101713
Issue: 149074 Duplicate Issue: 79112
Issue: 149423 Duplicate Issue: 145553
Issue: 149433 Duplicate Issue: 128422
Issue: 149464 Duplicate Issue: 148995
Issue: 149545 Duplicate Issue: 149440
Issue: 149640 Duplicate Issue: 193045
Issue: 149727 Duplicate Issue: 140559
Issue: 149752 Duplicate Issue: 92821
Issue: 149782 Duplicate Issue: 148373
Issue: 149795 Duplicate Issue: 92250
Issue: 149836 Duplicate Issue: 157055
Issue: 149853 Duplicate Issue: 129584
Issue: 149868 Dup

Issue: 162555 Duplicate Issue: 153913
Issue: 162568 Duplicate Issue: 83706
Issue: 162614 Duplicate Issue: 147673
Issue: 162772 Duplicate Issue: 99928
Issue: 162869 Duplicate Issue: 162743
Issue: 162876 Duplicate Issue: 163192
Issue: 162924 Duplicate Issue: 137398
Issue: 162978 Duplicate Issue: 134684
Issue: 163093 Duplicate Issue: 322817
Issue: 163129 Duplicate Issue: 158807
Issue: 163262 Duplicate Issue: 122610
Issue: 163354 Duplicate Issue: 157798
Issue: 163416 Duplicate Issue: 49619
Issue: 163417 Duplicate Issue: 29860
Issue: 163528 Duplicate Issue: 148872
Issue: 163560 Duplicate Issue: 106744
Issue: 163567 Duplicate Issue: 101476
Issue: 163679 Duplicate Issue: 61529
Issue: 163772 Duplicate Issue: 153055
Issue: 163829 Duplicate Issue: 162573
Issue: 163889 Duplicate Issue: 159607
Issue: 163935 Duplicate Issue: 162621
Issue: 163960 Duplicate Issue: 36350
Issue: 164065 Duplicate Issue: 164064
Issue: 164090 Duplicate Issue: 164091
Issue: 164156 Duplicate Issue: 157479
Issue: 164259 Dupl

Issue: 178585 Duplicate Issue: 254603
Issue: 178867 Duplicate Issue: 178185
Issue: 178969 Duplicate Issue: 178795
Issue: 178978 Duplicate Issue: 177910
Issue: 179039 Duplicate Issue: 159699
Issue: 179179 Duplicate Issue: 38016
Issue: 179195 Duplicate Issue: 305259
Issue: 179277 Duplicate Issue: 77451
Issue: 179392 Duplicate Issue: 177910
Issue: 179536 Duplicate Issue: 165533
Issue: 179690 Duplicate Issue: 179684
Issue: 179902 Duplicate Issue: 202404
Issue: 180119 Duplicate Issue: 177910
Issue: 180174 Duplicate Issue: 76435
Issue: 180381 Duplicate Issue: 167562
Issue: 180944 Duplicate Issue: 165223
Issue: 180947 Duplicate Issue: 22243
Issue: 180949 Duplicate Issue: 22243
Issue: 181059 Duplicate Issue: 181058
Issue: 181165 Duplicate Issue: 176419
Issue: 181357 Duplicate Issue: 73244
Issue: 181374 Duplicate Issue: 177910
Issue: 181459 Duplicate Issue: 178551
Issue: 181475 Duplicate Issue: 178551
Issue: 181547 Duplicate Issue: 122247
Issue: 181652 Duplicate Issue: 136215
Issue: 181900 Dupl

Issue: 198955 Duplicate Issue: 184298
Issue: 198967 Duplicate Issue: 145202
Issue: 199054 Duplicate Issue: 199023
Issue: 199147 Duplicate Issue: 155835
Issue: 199221 Duplicate Issue: 153079
Issue: 199223 Duplicate Issue: 151487
Issue: 199386 Duplicate Issue: 158008
Issue: 199392 Duplicate Issue: 138184
Issue: 199425 Duplicate Issue: 200067
Issue: 199452 Duplicate Issue: 176471
Issue: 199718 Duplicate Issue: 77892
Issue: 199729 Duplicate Issue: 199728
Issue: 199866 Duplicate Issue: 41255
Issue: 199972 Duplicate Issue: 188759
Issue: 199993 Duplicate Issue: 199991
Issue: 200003 Duplicate Issue: 198890
Issue: 200015 Duplicate Issue: 395071
Issue: 200017 Duplicate Issue: 196635
Issue: 200080 Duplicate Issue: 199668
Issue: 200200 Duplicate Issue: 169222
Issue: 200245 Duplicate Issue: 201758
Issue: 200454 Duplicate Issue: 63573
Issue: 200688 Duplicate Issue: 8112
Issue: 200817 Duplicate Issue: 236850
Issue: 200819 Duplicate Issue: 200818
Issue: 200929 Duplicate Issue: 200198
Issue: 201022 Dup

Issue: 220519 Duplicate Issue: 158526
Issue: 220788 Duplicate Issue: 190737
Issue: 220976 Duplicate Issue: 173714
Issue: 221516 Duplicate Issue: 221512
Issue: 221637 Duplicate Issue: 79798
Issue: 221764 Duplicate Issue: 189398
Issue: 221769 Duplicate Issue: 166223
Issue: 221785 Duplicate Issue: 221110
Issue: 222532 Duplicate Issue: 197733
Issue: 222694 Duplicate Issue: 102590
Issue: 222743 Duplicate Issue: 222392
Issue: 222814 Duplicate Issue: 68305
Issue: 222933 Duplicate Issue: 205235
Issue: 223060 Duplicate Issue: 77892
Issue: 223061 Duplicate Issue: 196140
Issue: 223124 Duplicate Issue: 158526
Issue: 223265 Duplicate Issue: 97912
Issue: 223334 Duplicate Issue: 216565
Issue: 223414 Duplicate Issue: 182360
Issue: 223448 Duplicate Issue: 214092
Issue: 223490 Duplicate Issue: 167952
Issue: 223497 Duplicate Issue: 149768
Issue: 223593 Duplicate Issue: 99933
Issue: 223743 Duplicate Issue: 222512
Issue: 223850 Duplicate Issue: 46447
Issue: 223868 Duplicate Issue: 184928
Issue: 223936 Dupl

Issue: 246637 Duplicate Issue: 243692
Issue: 246659 Duplicate Issue: 243692
Issue: 246685 Duplicate Issue: 243692
Issue: 246692 Duplicate Issue: 244716
Issue: 246715 Duplicate Issue: 246712
Issue: 246728 Duplicate Issue: 266582
Issue: 246816 Duplicate Issue: 32340
Issue: 246849 Duplicate Issue: 163554
Issue: 247178 Duplicate Issue: 243692
Issue: 247809 Duplicate Issue: 164449
Issue: 247848 Duplicate Issue: 243101
Issue: 247905 Duplicate Issue: 83651
Issue: 248080 Duplicate Issue: 244716
Issue: 248246 Duplicate Issue: 274898
Issue: 248303 Duplicate Issue: 249456
Issue: 248329 Duplicate Issue: 243179
Issue: 248537 Duplicate Issue: 170431
Issue: 248560 Duplicate Issue: 243692
Issue: 248628 Duplicate Issue: 239647
Issue: 248650 Duplicate Issue: 275308
Issue: 248781 Duplicate Issue: 245973
Issue: 249241 Duplicate Issue: 277574
Issue: 249425 Duplicate Issue: 21402
Issue: 249677 Duplicate Issue: 241155
Issue: 249726 Duplicate Issue: 249456
Issue: 249732 Duplicate Issue: 283950
Issue: 249756 D

Issue: 275801 Duplicate Issue: 255381
Issue: 275876 Duplicate Issue: 269182
Issue: 275918 Duplicate Issue: 194358
Issue: 276021 Duplicate Issue: 269182
Issue: 276847 Duplicate Issue: 183880
Issue: 277301 Duplicate Issue: 277812
Issue: 277302 Duplicate Issue: 277812
Issue: 277305 Duplicate Issue: 280668
Issue: 277432 Duplicate Issue: 38185
Issue: 277895 Duplicate Issue: 237090
Issue: 277897 Duplicate Issue: 236818
Issue: 277952 Duplicate Issue: 267178
Issue: 278146 Duplicate Issue: 341232
Issue: 278236 Duplicate Issue: 277697
Issue: 278749 Duplicate Issue: 261198
Issue: 278808 Duplicate Issue: 195834
Issue: 279402 Duplicate Issue: 248954
Issue: 279466 Duplicate Issue: 234769
Issue: 280054 Duplicate Issue: 277643
Issue: 280220 Duplicate Issue: 267616
Issue: 280298 Duplicate Issue: 266510
Issue: 280410 Duplicate Issue: 278745
Issue: 280550 Duplicate Issue: 250425
Issue: 280563 Duplicate Issue: 250425
Issue: 280809 Duplicate Issue: 277643
Issue: 280819 Duplicate Issue: 65876
Issue: 280951 

Issue: 313086 Duplicate Issue: 311863
Issue: 313212 Duplicate Issue: 251200
Issue: 313308 Duplicate Issue: 312867
Issue: 313737 Duplicate Issue: 58205
Issue: 313824 Duplicate Issue: 181788
Issue: 314073 Duplicate Issue: 139197
Issue: 314596 Duplicate Issue: 172256
Issue: 314931 Duplicate Issue: 313972
Issue: 315491 Duplicate Issue: 112433
Issue: 316090 Duplicate Issue: 289247
Issue: 316678 Duplicate Issue: 257313
Issue: 316782 Duplicate Issue: 189459
Issue: 317012 Duplicate Issue: 289247
Issue: 317105 Duplicate Issue: 283503
Issue: 317313 Duplicate Issue: 172256
Issue: 317367 Duplicate Issue: 298936
Issue: 317576 Duplicate Issue: 123836
Issue: 317762 Duplicate Issue: 317719
Issue: 317784 Duplicate Issue: 308000
Issue: 317829 Duplicate Issue: 321926
Issue: 317836 Duplicate Issue: 315732
Issue: 317893 Duplicate Issue: 309334
Issue: 318180 Duplicate Issue: 280801
Issue: 318243 Duplicate Issue: 313530
Issue: 318247 Duplicate Issue: 317719
Issue: 318276 Duplicate Issue: 168565
Issue: 318298

Issue: 355773 Duplicate Issue: 304176
Issue: 356018 Duplicate Issue: 353535
Issue: 356282 Duplicate Issue: 301990
Issue: 356303 Duplicate Issue: 342332
Issue: 356417 Duplicate Issue: 305172
Issue: 356678 Duplicate Issue: 356677
Issue: 356679 Duplicate Issue: 356677
Issue: 356949 Duplicate Issue: 352990
Issue: 358045 Duplicate Issue: 357918
Issue: 359569 Duplicate Issue: 311212
Issue: 360523 Duplicate Issue: 251824
Issue: 360600 Duplicate Issue: 359646
Issue: 361115 Duplicate Issue: 346175
Issue: 361181 Duplicate Issue: 348024
Issue: 362102 Duplicate Issue: 361916
Issue: 362154 Duplicate Issue: 212262
Issue: 362775 Duplicate Issue: 393192
Issue: 362969 Duplicate Issue: 334622
Issue: 363512 Duplicate Issue: 348507
Issue: 363556 Duplicate Issue: 359173
Issue: 363724 Duplicate Issue: 314556
Issue: 364008 Duplicate Issue: 361053
Issue: 364485 Duplicate Issue: 152568
Issue: 364862 Duplicate Issue: 320487
Issue: 365015 Duplicate Issue: 106228
Issue: 365455 Duplicate Issue: 346175
Issue: 36578

Issue: 415587 Duplicate Issue: 382069
Issue: 415600 Duplicate Issue: 410325
Issue: 415948 Duplicate Issue: 415803
Issue: 415993 Duplicate Issue: 329294
Issue: 416180 Duplicate Issue: 416174
Issue: 416766 Duplicate Issue: 415276
Issue: 416853 Duplicate Issue: 412076
Issue: 416917 Duplicate Issue: 410325
Issue: 417662 Duplicate Issue: 414038
Issue: 417757 Duplicate Issue: 417758
Issue: 417795 Duplicate Issue: 403749
Issue: 417796 Duplicate Issue: 403749
Issue: 418232 Duplicate Issue: 412076
Issue: 418233 Duplicate Issue: 415269
Issue: 418273 Duplicate Issue: 416267
Issue: 418726 Duplicate Issue: 395348
Issue: 418807 Duplicate Issue: 419048
Issue: 418827 Duplicate Issue: 416267
Issue: 418843 Duplicate Issue: 416267
Issue: 419121 Duplicate Issue: 416267
Issue: 419152 Duplicate Issue: 415803
Issue: 419183 Duplicate Issue: 248059
Issue: 419506 Duplicate Issue: 416911
Issue: 419578 Duplicate Issue: 416267
Issue: 419767 Duplicate Issue: 377883
Issue: 419821 Duplicate Issue: 419331
Issue: 41986

In [23]:
print("True Positive: " + str(tp))
print("False Negative: " + str(fn))

True Positive: 1295
False Negative: 6393


In [24]:
print("Recall: " +str(tp/(tp+fn)))

Recall: 0.1684443288241415


# FIREFOX
##### Vector size:200 Epoch:20 Recall: 0.07477522756463953 (True Positive: 2678 False Negative: 33136)
##### Vector size:300 Epoch:20 Recall: 0.07689730273077568 (True Positive: 2754 False Negative: 33060)
##### Vector size:300 Epoch:20 Recall: 0.08661417322834646 (True Positive: 3102 False Negative: 32712) (Component filtreli)
##### Vector size:300 Epoch:300 Recall: 0.07566873289774949 (True Positive: 2710 False Negative: 33104)
##### Vector size:400 Epoch:10 Recall: 0.06704082202490647 (True Positive: 2401 False Negative: 33413)
##### Vector size:400 Epoch:20 Recall: 0.07530574635617357 (True Positive: 2697 False Negative: 33117)

# THUNDERBIRD
##### Vector size:300 Epoch:20 Recall: 0.08383329333653308 (True Positive: 1048 False Negative: 11453) (Component filtreli)

# ECLIPSE_JDT
##### Vector size:300 Epoch:20 Recall: 0.1650624349635796 (True Positive: 1269 False Negative: 6419) (Component filtreli)
##### Vector size:300 Epoch:40 Recall: 0.20096253902185224 (True Positive: 1545 False Negative: 6143) (Component filtreli)
##### Vector size:300 Epoch:100 Recall: 0.2144901144640999 (True Positive: 1649 False Negative: 6039) (Component filtreli)

In [ ]:
import tensorflow as tf